cloud-reactor/src/main/java/com/sequenceiq/cloudbreak/cloud/handler/StopStartDownscaleStopInstancesHandler.java

```diff
@@ -0,0 +1,90 @@
+
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<?> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = getAuthenticatedContext(request, cloudContext, connector);
+
+            List<CloudInstance> cloudInstancesToStop = request.getCloudInstancesToStop();
+            List<CloudVmInstanceStatus> cloudVmInstanceStatusList = Collections.emptyList();
+            if (cloudInstancesToStop.size() > 0) {
+                LOGGER.info("Attempting to stop instances with a timebound of {}ms. count={}, instances=[{}]",
+                        STOP_POLL_TIMEBOUND_MS, cloudInstancesToStop.size(), cloudInstancesToStop.stream().map(CloudInstance::getInstanceId)
+                                .collect(Collectors.toList()));
+                // TODO CB-14929: Error handlnig. Currently - an error from the following command (e.g. NPE)
+                //  results in no WARNINGS on the console. The stop is just ignored, and the cluster goes into a
+                //  RUNNING state, with the nodes in a DECOMMISSIONED state. The error handling needs to make sure
+                //  that the system goes into a failed state, and there is at least some logging on the console
+                //  to indicate what happened.
+                // TODO CB-14929: Also, after re-provisioning nodes via CM, the CB UI does not seem to be refreshing the node
+                //  status to RUNNING, and instead the nodes stay in a STUCK state. Is the CM sync task not taking care of this?
+                // TODO CB-15132: What happens if the cloud provider does not know about an instance for which a STOP was requested. How does the API
+                //  behave.
+                cloudVmInstanceStatusList = connector.instances().stopWithLimitedRetry(ac, null, cloudInstancesToStop, STOP_POLL_TIMEBOUND_MS);
+            } else {
+                LOGGER.info("No cloud VM instances to stop. Succeeding flow step with no action taken");
+            }
+            LOGGER.trace("CloudVMInstanceStatusesPostStop={}", cloudVmInstanceStatusList);
+
+            // TODO CB-15132: If we fail to STOP all instances - one potential path for error handling would be to allow a subsequent upscale operation
+            //  to consider nodes which are in DECOMMISSIONED state, but RUNNING - as candidates for UPSCALE.
+            StopStartDownscaleStopInstancesResult result = new StopStartDownscaleStopInstancesResult(request.getResourceId(),
+                    request, cloudVmInstanceStatusList);
+            eventBus.notify(result.selector(), new Event<>(event.getHeaders(), result));
+        } catch (Exception e) {
+            // TODO CB-14929: Error handling. Any exceptions from here need to be handled properly. Moving the nodes into appropriate states,
+            //  failing the action. Potential trying to recover in a subsequent change.
+            throw e;
```

Comment:
do not do this. It is handled by flow engine, but it is not the proper behavior. this code was written in flow engine to handle badly written flows /flowhandler by mistake. you should notify and handle error properly. I think we should create something like ExceptionCatcherEventHandler for CloudPlatformEventHandler, I will create a jira about this for myself not to forget.

Reply:
This is planned. Let me see what I can do w.r.t the error handling explicitly, instead of relying on default error handling.
Did not include this, because the system seems to be working quite well with the defaults.

Reply:
To clarify ... I meant setting up explicit States for errors.
Guessing you mean a ExceptionCatcherEventHandler similar to ExceptionCatcherEventHandler which already exists but is not tailored towards CloudRequests.

Reply:
yes, but error handling is a must-have, we can not merge it in this way. Please check other CloudPlatformEventHandler implementations for example.

Reply:
I'm going to post a separate PR for this change. Did catch one issue while making the changes and testing more.

Reply:
Merged this PR. Resolving.